pmd-core/src/main/java/net/sourceforge/pmd/reporting/GlobalAnalysisListener.java

```diff
@@ -38,6 +38,16 @@
  */
 public interface GlobalAnalysisListener extends AutoCloseable {
 
+    /**
+     * Notify the implementation that the analysis is about to start,
+     * and how many files are to be processed.
+     * 
+     * @param totalFilesToAnalyze The total number of files to be analyzed.
+     */
+    default void startAnalysis(int totalFilesToAnalyze) {
```

Comment:
@oowekyala I decided against a separate interface to keep things concise, as we already have a "global" listener, it felt right to notify when starting how many files where collected.

Reply:
I don't know... I'm concerned about the extensibility of this thing. If we want to pass other information to the start callback later, then we will have to muddy this interface with new methods, and then it's unclear in what order they will be called or which one should be implemented. Just passing the total number of files also feels too specific to me. It's hard to imagine how this method would be relevant for anything else than a progress bar, yet its name is very general.

What do you think of something like that:
```java
interface GlobalAnalysisListener {
     // called exactly once before any call to startFileAnalysis
     default ListenerInitializer prepareAnalysis() { return Noop(); }
}

interface ListenerInitializer extends AutoCloseable {
    default void setNumberOfFilesToAnalyze(int totalFiles) { }
    // we can add other methods here later, eg to give the listener access to a cache directory or to a message reporter
}
```
It still uses GlobalAnalysisListener as the main listener type for simplicity, but I feel like it will be easier to evolve this interface.

Reply:
Sounds good